lib/rules/no-multiple-objects-in-class.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @author tyankatsu <https://github.com/tyankatsu0105>
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const { defineTemplateBodyVisitor } = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/**
+ * count ObjectExpression element
+ * @param {VAttribute} node
+ * @return {number}
+ */
+function countObjectExpression(node) {
+  return node.value.expression.elements.filter(
+    (element) => element.type === 'ObjectExpression'
+  ).length
+}
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow to pass multiple objects into array to class',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-multiple-objects-in-class.html'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      unexpected: 'Unexpected multiple objects. Merge objects.'
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    return defineTemplateBodyVisitor(context, {
+      /** @param {VAttribute} node */
+      'VAttribute[directive=true][key.argument.name="class"][value.expression.type="ArrayExpression"]'(
```

Comment:
Could you check if it is v-bind?
I don't think I'll write this, but there is probably a false positive in `v-on:class="[{},{}]"`.

Reply:
@ota-meshi 
I verified `v-on` and `v-bind`.
It seems that `v-on` is not checked.

```js
invalid: [
  {
    code: `<template><div v-bind:class="[{'foo': isFoo}, {'bar': isBar}]" /></template>`,
    errors: [
      {
        message: "Unexpected multiple objects. Merge objects.",
        type: "VAttribute"
      }
    ]
  },
  {
    code: `<template><div :class="[{'foo': isFoo}, {'bar': isBar}]" /></template>`,
    errors: [
      {
        message: "Unexpected multiple objects. Merge objects.",
        type: "VAttribute"
      }
    ]
  },
  {
    code: `<template><div v-on:class="[{'foo': isFoo}, {'bar': isBar}]" /></template>`,
    errors: [
      {
        message: "Unexpected multiple objects. Merge objects.",
        type: "VAttribute"
      }
    ]
  },
  {
    code: `<template><div @class="[{'foo': isFoo}, {'bar': isBar}]" /></template>`,
    errors: [
      {
        message: "Unexpected multiple objects. Merge objects.",
        type: "VAttribute"
      }
    ]
  }
];
```

```bash
    invalid
      ✓ <template><div v-bind:class="[{'foo': isFoo}, {'bar': isBar}]" /></template>
      ✓ <template><div :class="[{'foo': isFoo}, {'bar': isBar}]" /></template>
      1) <template><div v-on:class="[{'foo': isFoo}, {'bar': isBar}]" /></template>
      2) <template><div @class="[{'foo': isFoo}, {'bar': isBar}]" /></template>

1) no-multiple-objects-in-class
       invalid
         <template><div v-on:class="[{'foo': isFoo}, {'bar': isBar}]" /></template>:

      AssertionError [ERR_ASSERTION]: Should have 1 error but had 0: []
      + expected - actual

      -0
      +1

      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:666:24)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:883:25)
      at processImmediate (internal/timers.js:456:21)

  2) no-multiple-objects-in-class
       invalid
         <template><div @class="[{'foo': isFoo}, {'bar': isBar}]" /></template>:

      AssertionError [ERR_ASSERTION]: Should have 1 error but had 0: []
      + expected - actual

      -0
      +1

      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:666:24)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:883:25)
      at processImmediate (internal/timers.js:456:21)
```

This result is what I expected.

Reply:
@tyankatsu0105
I misunderstood. You're right In case of `v-on`, it is parsed to `VOnExpression`, so there is no problem.
However, since `v-foo` is parsed to `ArrayExpression`, there is a false positive.

```vue
<template><div v-foo:class="[{'foo': isFoo}, {'bar': isBar}]" /></template>
```

So we need to check if it is `v-bind`.

Reply:
@ota-meshi 
I added `[key.name.name="bind"]`👍